heartbeat/SMBinclude

```diff
@@ -0,0 +1,503 @@
+</parameter>
+
+<parameter name="guest_ok" unique="0" required="0">
+<longdesc lang="en">
+If this parameter is yes for a service, then no password is required to connect to the service. Privileges will be those of the guest account.
+</longdesc>
+<shortdesc lang="en">guest ok parameter</shortdesc>
+<content type="string" default="${OCF_RESKEY_guest_ok_default}" />
+</parameter>
+
+<parameter name="directory_mask" unique="0" required="0">
+<longdesc lang="en">
+This parameter is the octal modes which are used when converting DOS modes to UNIX modes when creating UNIX directories.
+</longdesc>
+<shortdesc lang="en">directory mask parameter</shortdesc>
+<content type="string" default="${OCF_RESKEY_directory_mask_default}" />
+</parameter>
+
+<parameter name="create_mask" unique="0" required="0">
+<longdesc lang="en">
+When a file is created, the necessary permissions are calculated according to the mapping from DOS modes to UNIX permissions,
+and the resulting UNIX mode is then bit-wise ANDed with this parameter. This parameter may be thought of as a bit-wise MASK for the UNIX modes of a file. 
+</longdesc>
+<shortdesc lang="en">create mask parameter</shortdesc>
+<content type="string" default="${OCF_RESKEY_create_mask_default}" />
+</parameter>
+
+<parameter name="printable" unique="0" required="0">
+<longdesc lang="en">
+If this parameter is yes, then clients may open, write to and submit spool files on the directory specified for the service.
+</longdesc>
+<shortdesc lang="en">printable parameter</shortdesc>
+<content type="string" default="${OCF_RESKEY_printable_default}" />
+</parameter>
+
+<parameter name="valid_users" unique="0" required="0">
+<longdesc lang="en">
+This is a list of users that should be allowed to login to this service. Names starting with @, + and & are interpreted 
+using the same rules as described in the invalid users parameter.
+</longdesc>
+<shortdesc lang="en">valid users parameter</shortdesc>
+<content type="string" default="${OCF_RESKEY_valid_users_default}" />
+</parameter>
+
+<parameter name="force_user" unique="0" required="0">
+<longdesc lang="en">
+This specifies a UNIX user name that will be assigned as the default user for all users connecting to this service. This is useful for sharing files.
+</longdesc>
+<shortdesc lang="en">force user parameter</shortdesc>
+<content type="string" default="${OCF_RESKEY_force_user_default}" />
+</parameter>
+
+<parameter name="force_group" unique="0" required="0">
+<longdesc lang="en">
+This specifies a UNIX group name that will be assigned as the default primary group for all users connecting to this service.
+This is useful for sharing files by ensuring that all access to files on service will use the named group for their permissions checking.
+</longdesc>
+<shortdesc lang="en">force group parameter</shortdesc>
+<content type="string" default="${OCF_RESKEY_force_group_default}" />
+</parameter>
+
+<parameter name="extraopt1" unique="0" required="0">
```

Comment:
I think we should make this extraopts and make the options comma, semi-colon or colon-separated.

Reply:
unfortunately not possible after these signs are used by samba.
I've hat a look in the smb.conf man page:

password server = windc.mydomain.com:389 192.168.1.101 *
read list = mary, @students
print command = lp -d%p -s %s; rm %s

If you think I could reduce to two extraopts, one which is fe. semi-colon seperated for having a list and one which still uses plain for having the possibility to use samba configurations as shown above.

Reply:
Ok, I've changed this and made 2 possibilities:
_extraopt_  ->  this can be a key value pair which could include a semicolon
_extraopt_list_ -> here you can put in a semicolon separated list of key value pairs. 

So I think this solves all needs.